oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryTraceHandler.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.receiver.otel.otlp;
+
+import com.google.gson.JsonObject;
+import com.google.protobuf.ByteString;
+import io.grpc.stub.StreamObserver;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceResponse;
+import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpc;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.InstrumentationScope;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.ScopeSpans;
+import io.opentelemetry.proto.trace.v1.Status;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.core.server.GRPCHandlerRegister;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+import org.apache.skywalking.oap.server.receiver.otel.Handler;
+import org.apache.skywalking.oap.server.receiver.sharing.server.SharingServerModule;
+import org.apache.skywalking.oap.server.receiver.zipkin.SpanForwardService;
+import org.apache.skywalking.oap.server.receiver.zipkin.ZipkinReceiverModule;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RequiredArgsConstructor
+public class OpenTelemetryTraceHandler
+    extends TraceServiceGrpc.TraceServiceImplBase
+    implements Handler {
+    private final ModuleManager manager;
+    private SpanForwardService forwardService;
+
+    @Override
+    public String type() {
+        return "otlp-traces";
```

Comment:
This is misleading users that we receive OpenTelemetry traces data and convert into SkyWalking format, please rename this to something like `otlp-traces-zipkin`

Reply:
I think `to Zipkin` should be an internal configuration, rather than a type? `oltp-traces` means we support `otel`'s trace format. Whether we forward that to Zipkin or SkyWalking trace(nearly 100% impossible` should be determined by the handler configuration, rather than hanlder's type.

Reply:
Nor are there handler configurations that indicate this is Zipkin format

Reply:
I know. My point is, if you prefer to add, add there.
And also, it is nearly impossible to support skywalking trace, I am afraid that would be a fact for a very long time, maybe even forever.

Reply:
> I know. My point is, if you prefer to add, add there.
> And also, it is nearly impossible to support skywalking trace, I am afraid that would be a fact for a very long time, maybe even forever.

I'm not arguing (whether it's possible) to add SkyWalking format support, I'm saying if it doesn't support (for now), make it clear for users.

Reply:
I know this is not an arguing. My point is, that we rarely declared anything from the `application.yml`.
And AFAIK, OTLP trace doc has been updated as

> SkyWalking can receive traces from OpenTelemetry Trace and convert it into Zipkin Trace storage.